sdk/rust/oak/src/lib.rs

```diff
@@ -524,9 +516,37 @@ pub fn run_command_loop<
     }
 }
 
+/// A trait that encapsulates details about the entrypoint of an Oak Node, in a type safe way.
+pub trait WasmEntrypoint {
+    /// The name of the Wasm exported function.
+    const EXPORT_NAME: &'static str;
+    /// The Type of messages that the Node expects to handle on its inbound channel.
+    type Message: crate::io::Encodable + crate::io::Decodable;
+}
+
+/// Creates a new Wasm node based on the provided [`WasmEntrypoint`] descriptor, including an
+/// inbound channel, and returns a typed [`crate::io::Sender`] to such channel.
+pub fn wasm_node_create<T: WasmEntrypoint>(
```

Comment:
An alternative approach here might be to define this as a macro too, that way users don't have to know the name of the type the other macro generates. A `wasm_node_create!` macro would only need to know the name of the entrypoint.